src/FastRandom.cpp

```diff
@@ -8,8 +8,11 @@
 
 #include <random>
 
-#if defined (__GNUC__)
-	#define ATTRIBUTE_TLS static __thread
+#if defined (ANDROID)
+	// Tycho: I'm sorry, I'm sorry, I'm so, so, sorry, but please blame the NDK developers for lack of, thread_local
+	#define ATTRIBUTE_TLS static
```

Comment:
Why not just replace cCriticalSection with a noop? It will offer a major performance boost on these low power devices? Because we need thread safety. Any instance of a race condition is grounds for the compiler to do whatever it feels like.